### ISSUE TITLE
Compiling C++ Examples Without CMake after version 19.17

### ISSUE THREAD

**Issue by wpmanoj** [davisking/dlib#2130]
Steps i install dlib in Visual studio -
extract dlib to disk. set up include path VS.
added  dlib/all/source.cpp
added dlib/external/libjpeg .cpp files
set up DLIB_JPEG_SUPPORT

this steps works with until version 19.17 but after 19.17 some errors are occurred.

please let us know right way to install dlib newer versions without cmake.

* **Where did you get dlib**: dlib.net version release
* **Platform**: 64
* **Compiler**: VS (142)

**Comment by davisking** [comment]
What are "some errors"?

**Comment by wpmanoj** [comment]
firstly "printf" unsafe error occurred then i add "_CRT_SECURE_NO_WARNINGS" then occured

Severity	Code	Description	Project	File	Line	Suppression State
Error	C4996	'unlink': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _unlink. See online help for details.	test	D:\c++\libraries\dlib-19.20\dlib\external\libjpeg\jmemname.c	233	

then i set as "comment" suggest 
//unlink(info->temp_name);
  remove(info->temp_name);

then this all errors occurred 

Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK2005	jpeg_mem_term already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_mem_init already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_open_backing_store already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_mem_available already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_free_large already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_get_large already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_free_small already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_get_small already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_mem_term already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_mem_init already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_open_backing_store already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_mem_available already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_free_large already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_get_large already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_free_small already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	jpeg_get_small already defined in jmemansi.obj	test	D:\c++\test\test\jmemansi.obj	1	
Error	LNK2005	main already defined in jpegtran.obj	test	D:\c++\test\test\jpegtran.obj	1	
Error	LNK2005	main already defined in jpegtran.obj	test	D:\c++\test\test\jpegtran.obj	1	
Error	LNK2005	main already defined in jpegtran.obj	test	D:\c++\test\test\jpegtran.obj	1	
Error	LNK2001	unresolved external symbol keymatch	test	D:\c++\test\test\jpegtran.obj	1	
Error	LNK2001	unresolved external symbol read_stdin	test	D:\c++\test\test\jpegtran.obj	1	
Error	LNK2001	unresolved external symbol write_stdout	test	D:\c++\test\test\jpegtran.obj	1	
Error	LNK1120	3 unresolved externals	test	D:\c++\test\x64\Release\test.exe	1

**Comment by wpmanoj** [comment]
close this issue because no support from developers.

**Comment by davisking** [comment]
It's not clear to me what the problem is.  These are warnings, not errors.  And they are not correct.  `unlink()` is a posix function and it isn't replaced by `_unlink`.  Moreover, jmemname.c isn't even built by dlib's cmake scripts.  So you shouldn't be building it either.

**Comment by davisking** [comment]
I also don't want to go changing the libjpeg code.  There is a copy of
libjpeg in dlib/external only because a long time ago I used to tell people
who wanted to use libjpeg to download it themselves and link it into their
application. But a huge number of users don't understand what "download
libjpeg and link it to your application" means or understand how to do
that.  So now dlib comes with a copy of libjpeg.  By default, dlib's build
scripts use the system copy of libjpeg.  However, if no system copy is
available cmake will automatically build and use the copy in
dlib/external.

So TLDR, if you don't like the copy of libjpeg in dlib/external you
shouldn't use it.  Use one you like.  But do note that what's in
dlib/external is an unmodified copy of libjpeg.

**Comment by Greendogo** [comment]
@wpmanoj If you didn't get all that, @davisking is saying you can exclude all the libjpeg files from your compilation that do not appear in the list inside the CMakeLists.txt file inside the dlib folder.  Just search for "external/libjpeg/", there is a long list of the only files you need from libjpeg.  Just fyi, you also don't need the .h or .dj files either.  JUST the ones in the CMakeLists.txt.

You still need the flag: ` _CRT_SECURE_NO_WARNINGS`
